cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.metadata;
+
+
+/**
+ * Exception which is thrown when a metadata operation fails
+ */
+public class MetadataException extends RuntimeException {
```

Comment:
why is it a RuntimeException?

Reply:
Rethinking client might be able to handle it and recover in some cases. Changed it Exception.